src/canvas/BlendMode.js

```diff
@@ -122,45 +122,39 @@ var BlendMode = new function() {
         // else if (Cs == 1) B(Cb, Cs) = 1
         // else B(Cb, Cs) = min(1, Cb / (1 - Cs))
         'color-dodge': function() {
-            dr = br === 0 ? 0 : sr === 255 ? 255 : min(255, 255 * br / (255 - sr));
-            dg = bg === 0 ? 0 : sg === 255 ? 255 : min(255, 255 * bg / (255 - sg));
-            db = bb === 0 ? 0 : sb === 255 ? 255 : min(255, 255 * bb / (255 - sb));
+			dr = min (255, br == 0 ? 0 : sr < 255 ? 255 * br / (255 - sr) : 255);
+			dg = min (255, bg == 0 ? 0 : sg < 255 ? 255 * bg / (255 - sg) : 255);
+			db = min (255, bb == 0 ? 0 : sb < 255 ? 255 * bb / (255 - sb) : 255);
```

Comment:
@sapics I'm not sure these changes are better. We want blend-modes to be as fast as possible, and function calls are usually slower than direct code.

Reply:
I think so too, performance is more important for the drawing library!
And blend-mode calculation is done for each pixel, thus, performance is much more important.
If you need just `Add blend modes` commit, I will re-push with force option to this PR!

Reply:
Performance is likely to be impacted in the order of about 1%. The function min() is an alias of Math.min() and as such an inbuilt function with very little impact. Is there any example of an animation which uses blend modes and requires high performance? In my application I might try out different blend modes on a more-or-less static set of images, so performance was never a question. But I did find that the form with simple comparisons (== instead of ===, use of <, just in case sr, sg, sb were out of range, for whatever reason) a little easier to compare with the equations of these blend modes in the literature.

Reply:
@SkepticaLee By creating millions of large paths (1000x1000 px) which have non-native blend mode, I guess that you can feel performance issue in blend mode :)